### DISCUSSION TITLE
Message "You are running the legacy version of PnP PowerShell"

### DISCUSSION THREAD

**Discussion by Tom-CK** [pnp/powershell#121] [Q&A]
I am using legacy version of SharePoint PNP Online in many scripts, which I write for customers (none IT).
Since a while whenever I run the command "Connect-PNPOnline" there is a message "You are running the legacy version of PnP PowerShell". I know that I am using the legacy version !

Is there a way to avoid that message ? I get many tickets from customers because of this message .....

**Comment by erwinvanhunen** [comment]
We understand it's annoying, but we wanted to make it really clear that we like people to move over to the new version. We will do one more release (in a few days from now) of the legacy edition which will allow you to turn off this message by setting an environment variable.  We realize we should have given that option from day one, but unfortunately it didn't make it into the release. Apologies for that.

**Reply by Tom-CK** [reply]
Thank you for taking care !

I would prefer an additional parameter like "-noWarning" instead of an environment variable.

I will use legacy as soon as possible. The problem is, that many of my scripts are running on machines of my customers. Those machines are running PowerShell Version 5 and I do not now when they will be updated.
As soon as I can ensure to have PS 7 I will update SharePoint PNP Online Powershell.

Please allow me one more question:

I am not very sure, but I think before when using "$var = Get-PNPConnection" there was NO ErrorMessage when no Connection was established. In case of no connection, $var was just null or empty, but there was no error.
Since you bring up the message "You are running the legacy version ...." there is also an error message when no connection is established.

I was using "$var = Get-PNPConnection" to test, if there is an active connection. if (! $var), I was then using Connect-PNPOnline.

I am not able to supress the error message, Even when using -ErrorAction SilentlyContinue or | Out-Null - there is still an Error:
"The current connection holds no SharePoint context ...."

**Reply by erwinvanhunen** [reply]
Actually, you can use the new module just fine in PowerShell 5. It's being built to support both PS7 and PS5.

The message is being written by the Connect-PnPOline cmdlet. This creates an 'in-memory' object that is being returned by Get-PnPConnection. The Get-PnPConnection cmdlet will however not establish a connection, so you will always have to issue Connect-PnPOnline. With the December release this means you get a message. 

The reason you cannot suppress the message by 'swallowing' it by means of Out-Null is that we write the message directly to the host (e.g. PowerShell) instead of to the pipeline. I'll make sure to change that behavior in the last release of the legacy edition.

**Reply by Tom-CK** [reply]
Okay, great !

Please make sure to do so in the new PnP.PowerShell-Module also. 
Same problem there: Get-PnPConnection always throws an exception - no chance to swallow it ...

Do you have an idea of when the new release will be available ?

**Reply by ToddKlindt** [reply]
It's out. Version 3.29.2101.0 supports this. I just tested this and it worked.

**Reply by Tom-CK** [reply]
You guys are great, thank you !

I've just tested this, there is no more message "You are running ..." when running Get-PnPConnection.
But there is still an Error when no connection was establisehd, even when storing the result into a variable and when using -ErrorAction Silently continue.

When scripting, I do not want to use Connect-PnpOnline in every function call again. I us $con = Get-PnPConnection and then only call Connect-PnPOnline when $con is null. Now I get an error when doing this.
How can I test if a Connection is already established ?

**Reply by AndersRask** [reply]
As a general recommendation, I suggest replacing all Write-Host with Write-Warning/Write-Output depending on severity. Writing to host is something I would only consider doing in a one-off script or similar, but in a framework this should be avoided IMO

**Reply by AndersRask** [reply]
http://www.jsnover.com/blog/2013/12/07/write-host-considered-harmful/
